src/adagrams.js

```diff
@@ -1,15 +1,142 @@
+//WAVE 1
 export const drawLetters = () => {
-  // Implement this method for wave 1
-};
+  const LETTER_POOL = {
+    A: 9,
+    B: 2,
+    C: 2,
+    D: 4,
+    E: 12,
+    F: 2,
+    G: 3,
+    H: 2,
+    I: 9,
+    J: 1,
+    K: 1,
+    L: 4,
+    M: 2,
+    N: 6,
+    O: 8,
+    P: 2,
+    Q: 1,
+    R: 6,
+    S: 4,
+    T: 6,
+    U: 4,
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1,
+  };
+
+  let letterBank = [];
+  let letterCopy = JSON.parse(JSON.stringify(LETTER_POOL));
```

Comment:
This works but you could also create a copy of the letter pool using the spread operator. This is to prevent the letter pool from being mutated. The time complexity of JSON.stringify is O(n), where n is the total size of all the values in the input object. This means that the complexity is linear with respect to the size of the object. The JSON.stringify method iterates over all the values in the object, so the larger the object, the longer it will take to stringify it. This works and is very readable however the time complexity is O( n * n ) which is 0(n^2) because the outer parsing will run a max of `n` times. The spread operator (…) has a constant time complexity, which means that its performance is not affected by the size of the input. This makes the spread operator a great choice for dealing with arrays and objects of any size. Additionally, the spread operator is a great way to easily create shallow copies of objects, as well as to concatenate multiple arrays into one.
  let letterPoolCopy = [...letterPool];